pkg/scheduler/objects/queue.go

```diff
@@ -1292,3 +1299,35 @@ func (sq *Queue) String() string {
 	return fmt.Sprintf("{QueuePath: %s, State: %s, StateTime: %x, MaxResource: %s}",
 		sq.QueuePath, sq.stateMachine.Current(), sq.stateTime, sq.maxResource)
 }
+
+func (sq *Queue) incRunningApps() {
+	sq.Lock()
+	defer sq.Unlock()
+	sq.runningApps++
+	if sq.parent != nil {
+		sq.parent.incRunningApps()
+	}
+}
+
+func (sq *Queue) decRunningApps() {
+	sq.Lock()
+	defer sq.Unlock()
+	sq.runningApps--
+	if sq.parent != nil {
+		sq.parent.decRunningApps()
+	}
+}
+
+func (sq *Queue) canRun() bool {
+	sq.RLock()
+	defer sq.RUnlock()
+
+	if sq.maxRunningApps == 0 {
+		return true
+	}
+	ok := sq.runningApps < sq.maxRunningApps
+	if sq.parent != nil {
```

Comment:
Just an idea:
If `ok` is false, it means that the limit is reached and printing a message is useful.
I suggest this change:

```
ok := sq.runningApps < sq.maxRunningApps
if !ok {
  log.Logger().Info("Maximum number of runnable apps reached", zap.String("queue", sq.QueuePath))
}
```

or maybe we can return earlier:
```
ok := sq.runningApps < sq.maxRunningApps
if !ok {
  log.Logger().Info("Maximum number of runnable apps reached", zap.String("queue", sq.QueuePath))
  return false
}
if sq.parent != nil {
  return sq.parent.canRun()
}
return true
```